CONTRIBUTING.md

```diff
@@ -20,14 +20,33 @@ composer check-platform-reqs
 composer update
 ```
 
-*Note: using Docker for contributing is strongly discouraged, as it requires [extra knowledge of composer internals](https://github.com/composer/composer/issues/9368#issuecomment-718112361).*
-
 Then you can start working with the code :+1:
 
 <br>
 
 Do you want to **contribute a failing test**? [This tutorial will sow you how](https://github.com/rectorphp/rector/blob/main/docs/how_to_add_test_for_rector_rule.md)
 
+### Docker runtime
+
+*Note: using Docker for contributing is strongly discouraged, as it requires [extra knowledge of composer internals](https://github.com/composer/composer/issues/9368#issuecomment-718112361).*
```

Comment:
@TomasVotruba I've just rebased my changes on top of main and resolved conflicts. I've moved this information to this section.

However I strongly disagree with this 😃 Issue that you mentioned here is applicable only to CI process, as @seldaek mentioned. It does NOT affect local development (and `COMPOSER_ROOT_VERSION` is set inside container anyway). The only required thing for using Docker runtime is Docker knowledge and you really should NOT discourage people from using it. Just let people decide if they want pollute local machine with libraries or use Docker.

Reply:
I've moved this after one contributor could not get through the `composer install` due to this :) 
I want to keep contributing simple and easy.

Reply:
So let's change it to discourage Docker usage only for people don't familiar with Docker. It does not have anything to Composer and Git since there's exact same repo inside container as on localhost (because it's mapped through volume). If developer knows how to use Docker properly there won't be problems with this approach.

PS. Can you point me to the `composer install` problem?

Reply:
Here: https://github.com/rectorphp/rector/issues/7086#issuecomment-1092531861

Reply:
[That's not a Docker runtime problem, that's developer's problem](https://github.com/rectorphp/rector/issues/7086#issuecomment-1166570839) (invalid usage, unfamiliarity with Docker etc.). I don't think he even executed that command in container.